rascaline/src/calculators/soap/radial_integral/spline.rs

```diff
@@ -161,4 +162,55 @@ mod tests {
             epsilon=delta, max_relative=1e-6
         );
     }
+
+    /// Check that the `with_accuracy` spline can be directly loaded into
+    /// `from_tabulated` and that both give the same result.
+    #[test]
+    fn accuracy_tabulated() {
+        let max_radial = 8;
+        let max_angular = 8;
+        let parameters = SoapRadialIntegralSplineParameters {
+            max_radial: max_radial,
+            max_angular: max_angular,
+            cutoff: 5.0,
+        };
+
+        let gto = SoapRadialIntegralGto::new(SoapRadialIntegralGtoParameters {
+            max_radial: parameters.max_radial,
+            max_angular: parameters.max_angular,
+            cutoff: parameters.cutoff,
+            atomic_gaussian_width: 0.5,
+        }).unwrap();
+
+        let spline_accuracy: SoapRadialIntegralSpline = SoapRadialIntegralSpline::with_accuracy(parameters, 1e-2, gto).unwrap();
+
+        // Serialize and Deserialize spline points
+        let spline_str = serde_json::to_string(&spline_accuracy.spline).unwrap();
+        let spline_loaded: Value = serde_json::from_str(&spline_str).unwrap();
+        let spline_points: Vec<SplinePoint> = serde_json::from_value(json!(spline_loaded["points"])).unwrap();
+
+        let spline_tabulated = SoapRadialIntegralSpline::from_tabulated(parameters, spline_points).unwrap();
+
+        let rij = 3.4;
+        let shape = (max_angular + 1, max_radial);
+
+        let mut values_accuracy = Array2::from_elem(shape, 0.0);
+        let mut gradients_accuracy = Array2::from_elem(shape, 0.0);
+        spline_accuracy.compute(rij, values_accuracy.view_mut(), Some(gradients_accuracy.view_mut()));
+
+        let mut values_tabulated = Array2::from_elem(shape, 0.0);
+        let mut gradients_tabulated = Array2::from_elem(shape, 0.0);
+        spline_tabulated.compute(rij, values_tabulated.view_mut(), Some(gradients_tabulated.view_mut()));
+
+        assert_relative_eq!(
+            values_accuracy, values_tabulated,
+            epsilon=1e-15, max_relative=1e-16
+        );
+
+        assert_relative_eq!(
+            gradients_accuracy, gradients_tabulated,
+            epsilon=1e-15, max_relative=1e-16
+        );
```

Comment:
If we want to check these are exactly the same, we could use `assert_eq!` here instead of the relative_eq

Reply:
I tried this before but, the values are not exactly the same...

Reply:
ok, fair enough